static/docs/user-guide/using-docker.md

```diff
@@ -0,0 +1,108 @@
+# Using Docker
+
+Currently we don't provide any official Docker image, however you can easily
+create it yourself.
+
+## Writing the Dockerfile
+
+We support several ways to install DVC, the two recommended ones are through
+`pip` and `conda`.
+
+You can base your image on the official Python one:
+```dockerfile
+FROM python
+
+# Remember to include the cloud providers that you plan to use,
+# for example: "dvc[s3]"
+RUN pip install dvc
+```
+
+Or use a conda based image:
+```dockerfile
+FROM continuumio/miniconda3
+
+RUN conda install dvc -y --quiet
+```
+
+For a recommended reading about writing Dockerfiles for Python applications,
+you can take a look at https://pythonspeed.com/docker/
+
+## Using Docker for your development environment
+
+Since `dvc` is a command line application, we have several recommendiations
+to enhance the experience:
+  * Make sure your _locale_ is set to UTF-8
+  * Install the `Zsh` shell and its respective autocompletion script.
+  * Install `less` so you can pipe the output of `dvc pipeline show`.
+  * Enable movements with `Ctrl + Arrow` by editing the _inputrc_ file.
+  * Use `pip install dvc` at the end of the file for better caching.
+  * Set your `WORKDIR` to the path where you are going to mount your project.
+
+If you are using Linux, remember to map your user to your container. This way,
+you make sure your files are still writtable. For more information:
+https://docs.docker.com/engine/security/userns-remap/
+
+### Dockerfile
+
+The following is a suggestion for a Dockerfile to use during development:
+
+```dockerfile
+FROM python
+
+# Use UTF-8 as the default locale
+ENV LANG=C.UTF-8
+
+# Better movement with [Ctrl + →] and [Ctrl + ←]
+ENV INPUTRC=/etc/inputrc
+
+RUN set -ex \
+  && echo '"\e[1;5C": forward-word' >> /etc/inputrc \
+  && echo '"\e[1;5D": backward-word' >> /etc/inputrc
+
+# Set working directory.
+# Mount your project under the same path.
+WORKDIR /usr/src
+
+# Install useful programs
+RUN apt-get update -y && apt-get install -y \
+      zsh \
+      less \
+      git
+
+#  Set Zsh as the default shell
+SHELL ["/usr/bin/zsh", "-c"]
+
+# Enable autocompletion scripts for DVC
```

Comment:
add additional comment on why this command is so complicated? may be link to the doc where it's described

Reply:
Right, @shcheklein , all the `zstyle` commands are for beautifying the completion menu.
I'll add a comment)